File: python/leetcode/246.py
class Solution:
    def isStrobogrammatic(self, nums):
        dic = {"0": "0", "1": "1", "6": "9", "8": "8", "9": "6"}
        l, r = 0, len(nums) - 1
        while l <= r:
            if nums[l] not in dic or nums[l] != dic[nums[r]]: return False
            l += 1
            r -= 1
        return True

File: cpp/leetcode/246.cpp
class Solution {
public:
    bool isStrobogrammatic(string num) {
        unordered_map<char, char> map{{'0', '0'}, {'1', '1'}, {'6', '9'}, {'8', '8'}, {'9', '6'}};
        int n = num.length();
        for (int l = 0, r = n - 1; l <= r; l++, r--)
            if (map.find(num[l]) == map.end() || map[num[l]] != num[r])
                return false; 
        return true; 
    }
};